dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/message/KafkaMessageServlet.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.fat.kafka.message;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Duration;
+import java.util.List;
+
+import javax.servlet.annotation.WebServlet;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.junit.Test;
+
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.AbstractKafkaTestServlet;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.SimpleKafkaReader;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.SimpleKafkaWriter;
+
+@WebServlet("/kafkaMessageTest")
+public class KafkaMessageServlet extends AbstractKafkaTestServlet {
+
+    @Test
+    public void testKafkaMessage() {
+        SimpleKafkaWriter<String> writer = kafkaTestClient.writerFor(KafkaMessageBean.CHANNEL_IN);
+        String value = "hello";
+        writer.sendMessage(value);
+
+        SimpleKafkaReader<String> reader = kafkaTestClient.readerFor(KafkaMessageBean.CHANNEL_OUT);
+        List<ConsumerRecord<String, String>> records = reader.waitForRecords(1, Duration.ofSeconds(2));
+
+        assertEquals(1, records.size());
+
+        ConsumerRecord<String, String> record = records.get(0);
+        assertEquals("OUT_null", record.key()); //original key defaults to null
```

Comment:
I'd prefer either setting the key on the original message, or generating the key completely in the processor method.

Reply:
Done